048-avoid-broker-restarts-when-in-recovery.md

```diff
@@ -1,20 +1,25 @@
 # Avoid broker restart when in log recovery state
+> Note: This proposal has been updated since the time it was merged.
 
 This proposal describes a solution for KafkaRoller's shortcoming with brokers in log recovery.
 
 ## Current situation
 
-The current logic in KafkaRoller:
+The following list outlines the high-level logic currently implemented in `KafkaRoller` for rolling pods:
 - Take a list of Kafka pods and reorder them based on the readiness.
-- Roll unready pods first without considering why it could be unready.
 - If the pod is the controller, roll it last.
-- Do not roll the pod, if it has an impact on the availability such as causing under replicated partitions.
-- If a Kafka pod does not become ready within the operational timeout (300000ms by default) after restarting, then it may force restart it without considering why it's taking a long time.
+- If pod is not ready before rolling, wait for it to become ready until the operation timeout reaches.
+- If pod is in a stuck state e.g CrashLoopBackOff, force restart the pod.
+- If pod is unschedulable, fail the reconcilation. 
+- If the AdminClient connection to pod failed, force restart the pod. 
+- Check if rolling the pod would have an impact on the availability such as causing under replicated partitions.
+- If rolling the pod does affect the availibility, then restart the pod and wait for pod readiness.
```

Comment:
A missing not here?

```suggestion
- If rolling the pod does not affect the availibility, then restart the pod and wait for pod readiness. Otherwise consider other pods for rolling first.
```